src/game/collision.ts

```diff
@@ -1,12 +1,22 @@
 import { Circle } from "../model/Cricle";
 import { GameState } from "../model/GameState";
 import { Rectangle } from "../model/Rectangle";
-import { SpriteType } from "../model/TypedSprite";
+import {SpriteType, TypedSprite} from "../model/TypedSprite";
 
 export function isCollision(circle: Circle, rectangle: Rectangle): boolean {
-  //TODO
+  let circleRadiusX = circle.x + circle.radius;
+  let circleRadiusY = circle.y + circle.radius;
+  return rectangle.x <= circleRadiusX && rectangle.y <= circleRadiusY
+      && (circle.x - rectangle.x) <= circle.radius
+      && Math.abs(rectangle.y - circleRadiusY) <= rectangle.height;
+}
+
+function killAngel(angel: TypedSprite, gameState: GameState) {
```

Comment:
I would propose a general "removeSprite" function that is the used. Both functions are functionally the same at the moment.

Reply:
I agree, fixed